connectors/sources/onedrive.py

```diff
@@ -320,17 +381,20 @@ async def paginated_api_call(self, url, params=None, fetch_size=FETCH_SIZE):
                 break
 
     async def list_users(self, include_groups=False):
+        header = None
         params = {
             "$filter": "accountEnabled eq true",
             "$select": "userPrincipalName,mail,transitiveMemberOf,id,createdDateTime",
         }
         if include_groups:
             params["$expand"] = "transitiveMemberOf($select=id)"
+        else:
+            params["$filter"] += " and assignedLicenses/$count ne 0&$count=true"
+            header = {"ConsistencyLevel": "eventual"}
```

Comment:
The purpose of this change was that, we need to avoid API calls for the users that do not have a license, as  OneDrive can only be used by licensed users.
Also, requesting files for such users would return a Bad Request stating `User's mysite url was not found`

Reply:
> The purpose of this change was that, we need to avoid API calls for the users that do not have a license, as OneDrive can only be used by licensed users.

Why it's not used when `include_groups == True`?

Reply:
That's because this header does not allow usage of $expand which is necessary for fetching `transitiveMemberOf` https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http

Reply:
TIL. So we don't need to worry about `users that do not have a license` when `include_groups == True`?

Reply:
Correct. the `include_groups == True` is used only during access control sync that does not incur successive delta API calls for every user.

Reply:
So, in summary, unlicensed users will be added to access_control_doc, which is expected since they are indeed a user, however we won't hit api calls to fetch their files as there aren't any files for them until they have a license

Reply:
Can you add comments to explain it. Thank you.